scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/ClientHello.java

```diff
@@ -148,33 +150,41 @@ private ClientHello(ProtocolVersion version, SecureRandom secureRandom, SessionI
 		} else {
 			this.sessionId = SessionId.emptySessionId();
 		}
-
-		// the supported groups
-		// TODO make list of supported groups configurable
-		SupportedGroup[] supportedGroups = SupportedGroup.getPreferredGroups().toArray(new SupportedGroup[]{});
-		this.extensions.addExtension(new SupportedEllipticCurvesExtension(supportedGroups));
-
-		// the supported point formats
-		List<ECPointFormat> formats = Arrays.asList(ECPointFormat.UNCOMPRESSED);
-		this.extensions.addExtension(new SupportedPointFormatsExtension(formats));
-
-		// the certificate types the client is able to provide to the server
-		if (supportedClientCertificateTypes != null && !supportedClientCertificateTypes.isEmpty()) {
-			CertificateTypeExtension clientCertificateType = new ClientCertificateTypeExtension(true);
-			for (CertificateType certificateType : supportedClientCertificateTypes) {
-				clientCertificateType.addCertificateType(certificateType);
+		
+		boolean supportedClientCertificate = supportedClientCertificateTypes != null
+				&& !supportedClientCertificateTypes.isEmpty();
+		boolean supportedServerCertificate = supportedServerCertificateTypes != null
+				&& !supportedServerCertificateTypes.isEmpty();
+
+		if (supportedClientCertificate || supportedServerCertificate) {
```

Comment:
IMHO we should make the inclusion of the extensions dependent on whether the supported cipher suites require a certificate exchange and/or whether they are ECC based.
For that, we should include the preferred cipher suites in the constructor instead of having a separate setter for adding them individually. There are not many places where `ClientHello.addCipherSuite()` is used ...